### ISSUE TITLE
Error while reading from specific dimensions of DataArray

### ISSUE THREAD

**Issue by adityachivu** [G-Node/nixpy#296]
I am storing a 4D matrix of type np.uint16 in a DataArray. When I try to read a subset of the data, I get the following error in some cases:

```
Traceback (most recent call last):
  File "/path/to/env/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2910, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-39-bc26bec09c5a>", line 23, in <module>
    img = data[0, :, 0, 0] # OSError
  File "/path/to/env/lib/python3.5/site-packages/nixio/data_array.py", line 137, in __getitem__
    self._read_data(raw, count, offset)
  File "/path/to/env/lib/python3.5/site-packages/nixio/pycore/data_array.py", line 82, in _read_data
    super(DataArray, self)._read_data(data, count, offset)
  File "/path/to/env/lib/python3.5/site-packages/nixio/pycore/data_array.py", line 28, in _read_data
    dataset.read_data(data, count, offset)
  File "/path/to/env/lib/python3.5/site-packages/nixio/pycore/h5dataset.py", line 58, in read_data
    self.dataset.read_direct(data, sl)
  File "/path/to/env/lib/python3.5/site-packages/h5py/_hl/dataset.py", line 657, in read_direct
    self.id.read(mspace, fspace, dest, dxpl=self._dxpl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5d.pyx", line 181, in h5py.h5d.DatasetID.read
  File "h5py/_proxy.pyx", line 130, in h5py._proxy.dset_rw
  File "h5py/_proxy.pyx", line 84, in h5py._proxy.H5PY_H5Dread
OSError: Can't read data (src and dest data spaces have different sizes)
```

The following is an example for a NIX file created with DataArray of size `[5, 10, 15, 20]` The same Errors occur:

```python
import nixio as nix
import numpy as np
import os

file_name = 'nix_test.hd'

if not os.path.isfile(file_name):

    nf = nix.File.open(file_name, nix.FileMode.Overwrite)
    block = nf.create_block('test', 'test')
    data_arr = block.create_data_array('test', 'test', 
                                        data=np.random.randint(65000, size=[5, 10, 15, 20]))
    nf.close()

nf = nix.File.open(file_name, nix.FileMode.ReadOnly)

data = nf.blocks['test'].data_arrays['test']
print(data, data.shape)

# img = data[0, 0, 0, 0]
# img = data[0, 0, 0, :]
# img = data[0, 0, :, 0]
# img = data[0, 0, :, :]
img = data[0, :, 0, 0] # OSError
# img = data[0, :, 0, :] # OSError
# img = data[0, :, :, 0]
# img = data[0, :, :, :]
# img = data[:, 0, 0, 0] # OSError
# img = data[:, 0, 0, :] # OSError
# img = data[:, 0, :, 0] # OSError
# img = data[:, 0, :, :] # OSError
# img = data[:, :, 0, 0] # TypeError: Can't broadcast (5, 10, 1, 1) -> (5, 10, 1)
# img = data[:, :, 0, :] # TypeError: Can't broadcast (5, 10, 1, 20) -> (5, 10, 20)
# img = data[:, :, :, 0]
# img = data[:, :, :, :]

print(img.shape)
print('done')
nf.close()
```

Further, when I specify `backend = 'hdf5'` I get the following error, even though I followed the NIX cpp lib installation as well as python bindings installation. I tried uninstalling and reinstalling everything as well, but get the same error:

```
Traceback (most recent call last):
  File "/path/to/env/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2910, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-40-873383e93563>", line 14, in <module>
    nf = nix.File.open(file_name, nix.FileMode.ReadOnly, backend='hdf5')
  File "/path/to/env/lib/python3.5/site-packages/nixio/pycore/file.py", line 160, in open
    raise RuntimeError("HDF5 backend is not available.")
RuntimeError: HDF5 backend is not available.
```

**Comment by achilleas-k** [comment]
Thanks for the report. I'll have a good look into the indexing issue.

As for the missing HDF5 backend, I'm not entirely sure what the issue might be there. One thing you can do is during installation of NIXPy add `--with-nix` (so the full command would be `pip install nixio --install-option --with-nix`). This will force the (optional) bindings to be built and will produce an error if it fails. If it succeeds, the installation time should be noticeable longer than normal.

**Comment by adityachivu** [comment]
The modified pip statement worked. However, I had to reinstall the boost libraries to make it work. I kept getting a "boost for python3.5 not found" error.

Thanks again!

**Comment by achilleas-k** [comment]
Ah yes, good old boost.

I figured out the indexing issue and the fix is now in PR #297. Give it a try if you like. I'll have an official release for it soon.
